// pakkumised/lib/models/www.minuvalik.ee.js
var AbstractModel = require("abstract.model");

module.exports = {
        "pictures": {
            "main": function () {
                var result = [];

                this._$('#form_block table').eq(1).find('td').eq(0).children('a').children('img').each(function () {
                    result.push(this._$(this).attr('src'));
                })

                return result;
            }, 
            "other": function () {
                var result = []

                $('#form_block > div').eq(2).children('div').eq(0).children('img').each(function() {
                    result.push($(this).attr('src'))
                })

                return result
            }
        }, 
        "price": {
            discount: $('#form_block table').eq(1).find('td').eq(1).children('div').children('div').eq(0).text(),
            regular: $('#form_block table').eq(1).find('td').eq(1).children('div').children('div').eq(6).text(),
            percent: $('#form_block table').eq(1).find('td').eq(1).children('div').children('div').eq(7).text(),
            benefit: $('#form_block table').eq(1).find('td').eq(1).children('div').children('div').eq(8).text()
        }
        , "title": {
            "full": function() {
                $('#form_block > div').eq(0).find('span.red').remove()
                return $('#form_block > div').eq(0).text()
            }(),
            "short" : ""
        }
        , "seller": {
            info: $('#form_block > div').eq(1).html()
        }
        , "description": {
            "full": function() {
                $('#form_block > div').eq(2).children('div').eq(0).children('img').remove()
                return $('#form_block > div').eq(2).children('div').eq(0).html();
            }()
            , "short": ""
            , "map": $('#form_block iframe').attr('src')
        }
}
